action.yml

```diff
@@ -199,7 +199,7 @@ runs:
         fi
         # Update GitHub status description to 'Build started'
         curl -X POST \
-          -H "Authorization: Bearer ${{ env.GITHUB_TOKEN }}" \
+          -H "Authorization: Bearer ${{ inputs.github_token }}" \
```

Comment:
I still don't follow, whether this approach is fine.

Reply:
But based on this article https://dev.to/github/the-githubtoken-in-github-actions-how-it-works-change-permissions-customizations-3cgp it seems like we can use both approaches.

Reply:
Me too, I just followed what was already started.

However, from all of the available options, this one seems like the best one to me (even though it is not perfect).

We have the following options (as I see it):
1) pass GitHub token as ENV 
    - this is not transparent for a user, because setting ENV is (IMO) not possible to enforce, and when the user does not set it, it can be difficult to debug why the action failed
2) use ${{ github.token }} directly in composite
    - this is not documented anywhere and is close to impossible to test, so I do not know if it even would work. If this works, it could be a good solutions I think.
    - Negative for this is, that this too is not transparent for the user, as his token is used without his knowledge.
3) pass GitHub token as an input
    - this adds one more input, and we already have a lot of inputs
    - but positive is, that also on GitHub pages, they pass the GitHub token like this [1]
    - positive also is, that this is transparent for the user
   
But it is time for the decision I would say.

WDYT?

[1] https://docs.github.com/en/actions/security-guides/automatic-token-authentication#example-1-passing-the-github_token-as-an-input

Reply:
@fernflower 

I would like to hear also your opinion, as you added the github_token input. WDYT?

Reply:
I agree, that if the user missed adding to environment GITHUB_TOKEN, then updating the pull request will fail. This is not a good solution, I agree.

For the second case, it seems, like our GitHub Action can take `${{ github.token }}` variable which should be propagated. I am scared, if GitHub changes it, then we will not know, where is the problem.

For the third case, It is transparent, but another variable. Upstream projects which already use it will be broken.

What about adding logic into action, like:
if env is present, then use it. If the input variable is present, then use this one. And even if both are not present then use `${{ github.token }}`.

Reply:
I would agree with the proposal if we drop the ENV option at all.

So let's make `github_token` optional, with the default value being `${{ github.token }}`.

Reply:
This needs to be definitely tested.